### ISSUE TITLE
REF: separate exposure from offset

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#2175]
Offset is a generic, fixed additive term that can be added to any model that has a linear core (exog * params)
Exposure is currently we treated essentially the same way as offset after taking logs, and is only supported if the link function is `log`, i.e. exponential mean model.

I'm getting convinced over the last few times that I looked at it, that we should treat exposure as a generic multiplicative term for any link function, which only coincides with offset in the log-link case.

**Comment by josef-pkt** [comment]
@kshedden @jseabold 
long/medium term plan:  review and refactor offset exposure

currently milestone 0.8

**Comment by josef-pkt** [comment]
Exposure does not enter the variance function in the right way. This affects for example Tweedie. 
Exposure `t` should represent the sum of `t` independently and identically distributed random variables. This means the variance should be linear in t. Currently, log exposure is treated as an offset and added to mu if link is log-link. However, several functions, all except for Poisson, don't have variance linear in mu. 

see discussion starting here:
https://groups.google.com/d/msg/pystatsmodels/GsumngVNEnA/lhuUIeFBAAAJ

This is currently a bug for the interpretation of exposure. exposure right now is just a transformed offset with log-link restriction.
In my interpretation, exposure in GLM NegativeBinomial should also not behave in the right way because variance is quadratic in mu.

**Comment by lorentzenchr** [comment]
AFAICS exposure should be replaced/renamed by var_weights (in GLM context). If you have Z = Y/w (example, Y=counts, w=observation time, Z=frequency), then w is usually called exposure and is nothing else than a var_weight, i.e. Var[Z] ~ 1/w.
Therefore, as mentioned above, w, in general, cannot be modelled/implemented by a term in the linear predictor as it results in the wrong variance.

**Comment by josef-pkt** [comment]
My guess is that we focus on var_weights for now, but keeping an `exposure` keyword will be convenient when we don't want to rescale y as a user, e.g. exposure is more explicit if we have count distributions.

I also think now that the number of trials in binomial could by subsumed under exposure. var_weights can be used as an alternative, but the relationship is not "obvious" to non-experts.

**Comment by josef-pkt** [comment]
another usability/interpretation advantage of using exposure for the sum of independent random variables is that it has a proper distribution and not a translated distribution like the rescaled z = y / w

e.g. `poisson.pmf(y/t, lambda)` is not valid if y/t is not an integer (in terms of the actual distribution, not as extended quasi-likelihood), while `poisson.pmf(y, lambda * t)` is for t is exposure

**Comment by lorentzenchr** [comment]
Yes, the number of trials n in the Binomial case is a var_weight and one actually estimates the probability of one trial p=E[Z]=E[Y/n]. Z might not be Binomial/Poisson, but the distribution according to the exponential family - with var_weight w (=trials or time or other exposure) - is a completely valid pdf, called Bernoulli if Y is Binomial and might be called scaled Poisson if Y is Poisson.

**Comment by josef-pkt** [comment]
@lorentzenchr  the question is whether this is clear to average users? It wasn't clear to me a year ago.

our binomial family hides the underlying proportion rescaling, while R doesn't or gets into some difficulties explaining what the weights are doing.

The other property is just a implementation convenience, when we have a proper scipy.stats.distribution in `get_distribution` method for the implied distribution.
I started to think of Binomial as a count distribution that assumes iid trials, so we don't get overdispersion and failure of correct aggregation.
That it's also the distribution for proportions is partially a separate issue.

(In the underlying exponential family it will be equivalent, or different representation of the same distribution/sample, but superficially
